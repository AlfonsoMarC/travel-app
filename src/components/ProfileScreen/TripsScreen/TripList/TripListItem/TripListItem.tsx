import React, { useState } from "react";
import styled from "styled-components";
import { formatDate } from "helpers/dates";
import { Trip } from "types/types";
import { Link } from "react-router-dom";
import { small } from "assets/mixins";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Popover from "react-bootstrap/Popover";
import Button from "react-bootstrap/Button";

const TripCardContainer = styled.li`
  position: relative;
  min-width: 0;
  a {
    display: flex;
    flex-direction: column;
    flex-shrink: 0;
    flex-grow: 0;
    max-height: 200px;
    overflow: hidden;
    color: ${({ theme }) => theme.color.black};
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    border-radius: 5px;
    &:hover {
      .photo {
        transform: scale(1.3);
      }
    }
  }
  .meta {
    flex-shrink: 0;
    z-index: 0;
    position: relative;
    height: 100px;
  }
  .photo {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-size: cover;
    background-position: center;
    transition: transform 0.2s;
  }
  .text {
    display: flex;
    flex-direction: column;
    position: relative;
    z-index: 1;
    padding: ${({ theme }) => theme.spacing.space};
    background: ${({ theme }) => theme.color.white};
    overflow: hidden;
    &:before {
      content: "";
      position: absolute;
      left: -10px;
      top: 0;
      bottom: 0;
      z-index: -1;
      transform: skewX(-3deg);
      width: ${({ theme }) => theme.spacing.space3};
      background: ${({ theme }) => theme.color.white};
    }
  }
  h3 {
    display: flex;
    font-size: 12px;
  }
  .trip-dates {
    display: flex;
    ${small};
  }
  .description {
    overflow: hidden;
  }
  ${({ theme }) => theme.media.atMedium} {
    width: 200px;
    a {
      flex-direction: row;
      height: 72px;
      width: 200px;
      &:hover {
        .photo {
          transform: scale(1.3) rotate(3deg);
        }
      }
    }
    .meta {
      flex-basis: 40%;
    }
    .text {
      flex-basis: 60%;
      justify-content: space-between;
      overflow: unset;
    }
    .description {
      display: none;
    }
    &:nth-child(even) {
      a {
        flex-direction: row-reverse;
        .text {
          &:before {
            left: inherit;
            right: -10px;
            transform: skew(3deg);
          }
        }
      }
      h3 {
        justify-content: flex-end;
        text-align: end;
        max-height: 42px;
      }
      .trip-dates {
        justify-content: flex-end;
      }
    }
  }
`;

/* const Icon = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  font-family: "Material Icons";
  font-size: 24px;
  cursor: pointer;
`;
 */
interface Props {
  trip: Trip;
}

const TripListItem: React.FC<Props> = ({ trip }) => {
  const startDate = formatDate(trip.startDate, "MMM yyyy");

  const popover = (
    <Popover id="popover-basic">
      <Popover.Body>
        And here's some <strong>amazing</strong> content. It's very engaging.
        right?
      </Popover.Body>
    </Popover>
  );

  return (
    <TripCardContainer className="alt">
      <Link to={`../trip/${trip._id}`}>
        <div className="meta">
          <div
            className="photo"
            style={{
              backgroundImage: `url(${
                trip.posts?.[0]?.img ??
                "https://storage.googleapis.com/chydlx/codepen/blog-cards/image-2.jpg"
              })`
            }}
          />
        </div>
        <section className="text">
          <h3>{trip.title}</h3>
          {startDate && (
            <div className="trip-dates">
              <time>{startDate}</time>
            </div>
          )}
          {trip.description && (
            <p className="description">{trip.description}</p>
          )}
        </section>
      </Link>
      {/*    <OverlayTrigger
        trigger="click"
        placement="right"
        overlay={popover}
        rootClose={true}
      >
        <Icon>arrow_drop_down</Icon>
      </OverlayTrigger> */}
    </TripCardContainer>
  );
};

export default TripListItem;
