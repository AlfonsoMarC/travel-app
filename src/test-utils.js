import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import theme from "assets/theme";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import {
  faFeather,
  faMapMarkerAlt,
  faPlus,
  faImage,
  faTrash,
  faEllipsisH
} from "@fortawesome/free-solid-svg-icons";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import { store } from "./store/index";

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  library.add(
    fab,
    faFeather,
    faMapMarkerAlt,
    faPlus,
    faImage,
    faTrash,
    faEllipsisH
  );
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

const customRender = (ui, { route = "/" } = {}, options) => {
  window.history.pushState({}, "Test page", route);
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
