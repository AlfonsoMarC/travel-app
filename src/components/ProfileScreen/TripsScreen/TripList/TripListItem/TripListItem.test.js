import { Router, useLocation } from "react-router-dom";
import { render } from "test-utils";
import { createMemoryHistory } from "history";
import TripListItem from "./TripListItem";

describe("TripListItem", () => {
  const history = createMemoryHistory();
  //  const location = useLocation();

  test("should", () => {
    const testTrip = {
      _id: "test_id",
      title: "Test trip",
      startDate: "01/01/2000",
      endDate: "01/01/2000",
      posts: [{ img: "test_url" }]
    };
    const { container } = render(
      <Router location={history.location} navigator={history}>
        <TripListItem trip={testTrip} />
      </Router>
    );
    expect(container).toMatchSnapshot();
    console.log("container", container);
  });
});
