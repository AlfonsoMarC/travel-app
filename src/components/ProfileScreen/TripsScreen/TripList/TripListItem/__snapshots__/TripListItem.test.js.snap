// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripListItem should 1`] = `
<div>
  <li
    class="sc-bczRLJ jfdduE alt"
  >
    <a
      href="/trip/test_id"
    >
      <div
        class="meta"
      >
        <div
          class="photo"
          style="background-image: url(test_url);"
        />
      </div>
      <section
        class="text"
      >
        <h3>
          Test trip
        </h3>
        
      </section>
    </a>
  </li>
</div>
`;
